package.json

```
@@ -8,7 +8,7 @@
   },
   "scripts": {
     "start": "webpack-dev-server",
-    "start-https": "webpack-dev-server --https",
+    "start-https": "webpack-dev-server --https  --mode development",
```

my understanding is that this should be `--mode=development`. And it's only necessary if the webpack config's `mode` property is `production`.

```suggestion
    "start-https": "webpack-dev-server --https  --mode=development",
```

---

See related https://webpack.js.org/api/cli/#common-options

---

Oh yeah I added that because I was having this problem where, when I had esm-module-config installed via `npm link`, all errors (including in tests and the webpack-dev-server) referred to minified code. I mitigated the problem by installing esm-module-config from npm.

---

In order to work on MF-74, I've linked esm-module-config from react-root-decorator. I'm again getting minified traces:

```
 FAIL  src/openmrs-react-root-decorator.test.js                                    [70/70]
  ● Test suite failed to run                                                             
                                                                                         
    ReferenceError: System is not defined                                                
                                                                                         
    > 1 | System.register([],(function(t){return{execute:function(){t(function(t){var n={}
;function r(e){if(n[e])return n[e].exports;var u=n[e]={i:e,l:!1,exports:{}};return t[e].ca
...etc...
```

Have you seen this problem before, do you know how to fix it?

---

`npm link` is not how you do local development for this. The published version of the code is in `System.register()` format, which is not usable by webpack nor by the tests. Webpack is already configured to exclude esm-module-config as an external. However, for the tests you need to mock the esm-module-config module.

To see an example of how to mock a module in jest, see [here](https://github.com/openmrs/openmrs-esm-patient-chart/blob/master/jest.config.json#L7) and [here](https://github.com/openmrs/openmrs-esm-patient-chart/blob/master/__mocks__/openmrs-esm-api.mock.tsx)

---

Ah ok, that's clear now, thanks!